File: app/src/main/java/org/tensorflow/lite/examples/poseestimation/database/User/UserDao.java
package org.tensorflow.lite.examples.poseestimation.database.User;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@androidx.room.Dao
public interface UserDao {

    @Insert
    void setInsert(User DB);

    @Update
    void setUpdate(User DB);

    @Delete
    void setDelete(User DB);

    @Query("SELECT * FROM User") // 조회 쿼리
    List<User> getAll();

}

File: app/src/main/java/org/tensorflow/lite/examples/poseestimation/database/UserDB/UserDao.kt
package org.tensorflow.lite.examples.poseestimation.database.UserDB

import androidx.room.*

@Dao
interface UserDao {
    @Insert
    suspend fun create(userSchema: UserSchema): Long

    @Update
    suspend fun update(userSchema: UserSchema)

    @Delete
    suspend fun delete(userSchema: UserSchema)

    @Query("SELECT * FROM table_user")
    suspend fun readAll(): List<UserSchema>

    @Query("DELETE FROM table_user")
    suspend fun deleteAll()
}